tests/interfaces/test_gen_object_detector_blackbox_sal.py

```diff
@@ -0,0 +1,246 @@
+import unittest.mock as mock
+import numpy as np
+import pytest
+
+from smqtk_detection import DetectImageObjects
+
+from xaitk_saliency.interfaces.gen_object_detector_blackbox_sal import GenerateObjectDetectorBlackboxSaliency
+from xaitk_saliency.exceptions import ShapeMismatchError
+
+
+def test_generate_checks_success() -> None:
+    """
+    Tests successful passage though the wrapper method.
+    """
+    m_impl = mock.Mock(spec=GenerateObjectDetectorBlackboxSaliency)
+    # mock implementation result, first dim value doesn't matter here.
+    m_impl._generate.return_value = np.empty((3, 256, 256))
+
+    m_detector = mock.Mock(spec=DetectImageObjects)
+
+    # test reference detections inputs with matching lengths
+    test_bboxes = np.ones((5, 4), dtype=float)
+    test_scores = np.ones((5, 3), dtype=float)
+    test_objectness = np.ones((5,), dtype=float)
+
+    # 2-channel image as just HxW should work
+    test_image = np.ones((256, 256), dtype=np.uint8)
+    GenerateObjectDetectorBlackboxSaliency.generate(
+        m_impl,
+        test_image,
+        test_bboxes,
+        test_scores,
+        m_detector,
+    )
```

Comment:
It seems like we should be checking after each of these calls for something along the lines of `m_impl._generate.assert_called_with(...)`.